iceoryx2-bb/log/src/lib.rs

```diff
@@ -185,6 +188,34 @@ pub enum LogLevel {
     Fatal = 5,
 }
 
+impl LogLevel {
+    fn from_str_fuzzy(s: &str) -> LogLevel {
+        match s.to_lowercase().as_str() {
+            "trace" => LogLevel::Trace,
+            "debug" => LogLevel::Debug,
+            "info" => LogLevel::Info,
+            "warn" => LogLevel::Warn,
+            "error" => LogLevel::Error,
+            "fatal" => LogLevel::Fatal,
+            _ => {
+                println!("Error: unknown logging level {:?}", s);
+                println!("Warning: setting log level as : Info");
+                DEFAULT_LOG_LEVEL
+            }
+        }
+    }
+}
+
+/// Sets the log level by reading environment variable "IOX2_LOG_LEVEL"
+pub fn init_log_level() {
```

Comment:
What do you think about renaming this to `init_log_level_from_env_or_default` to make it more clear what happens. Since there is already a `set_log_level` function, it might even make sense to call this `set_log_level_from_env_or_default` to make it more consistent. Finally, there could be a `set_log_level_from_env_or(log_level)` function which combines the two and set's the log_level only when the env variable is not set.

Reply:
@elBoberido Done and rebsed my branch to main to squash the commits for a cleaner histroy